### PR TITLE
WT-11555 Send remap as a boolean to detect a race

### PR DESCRIPTION
**Pull request by sueloverso**
@keitharnoldsmith after all the comments on the other PR, I reviewed my change and came up with this one that I believe is cleaner and simpler. It passes early testing. I am going to let it run much, much longer.

### CODE REVIEW

**Comment by sueloverso** [comment]
@keitharnoldsmith this is now ready for review.

File: src/os_posix/os_fs.c [hunk 1]
```diff
@@ -698,7 +697,8 @@ __posix_file_write_mmap(
     if (pfh->mmap_buf != NULL && !pfh->mmap_resizing && pfh->mmap_size < offset + (wt_off_t)len)
         /* If we are actively extending the file, don't remap it on every write. */
         if ((remap_opportunities++) % WT_REMAP_SKIP == 0) {
-            __wt_prepare_remap_resize_file(file_handle, wt_session);
+            __wt_prepare_remap_resize_file(file_handle, wt_session, offset + (wt_off_t)len, &remap);
```
Comment (sueloverso) [inline src/os_posix/os_fs.c:700]: Alternatively here, I could make the prepare function take a NULL boolean and get rid of the new local var here.

File: src/os_posix/os_fallocate.c [hunk 2]
```diff
@@ -20,28 +20,27 @@
...
+        WT_SYSCALL_RETRY(op, ret);                                                  \
+        if (remap) {                                                                \
+            if (ret == 0)                                                           \
+                __wt_remap_resize_file(file_handle, wt_session);                    \
+            else {                                                                  \
+                __wt_release_without_remap(file_handle);                            \
+                WT_RET(ret);                                                        \
+            }                                                                       \
+        }                                                                           \
+        return (0);                                                                 \
```
Comment (keitharnoldsmith) [inline src/os_posix/os_fallocate.c:43]: Does this need to be `return (ret)`. It's not related to your change specifically. But it seems like a potential bug.

If we're not using a memory mapped file and the `fallocate` call fails, it seems we should return the error to the caller. Specifically, `__wt_posix_file_extend` seems to rely on this to determine which `fallocate` method it should use.

Likewise about the similar return from `__posix_file_truncate`, below.

**Review by keitharnoldsmith — APPROVED** [review]
LGTM.

Nice Sue, I like this approach!

In addition to simplifying this change, it has the additional benefit, I think, of moving knowledge of how memory mapped files are implemented out of generic code and keeping it in functions (`*_remap_*`) that specifically manipulate the mappings.

I have one question/comment about the return value in cases where we truncate a file that isn't memory mapped. It's a simple change, so no need for me to review. I'm also happy to discuss, or move it to a separate ticket if you think that would be clearer.

File: src/os_posix/os_fallocate.c (hunk 2 above)
Comment (sueloverso) [reply src/os_posix/os_fallocate.c:43]: Yes, that appears to be a potential bug. I'll look at this and change here too.

**Pull request merged**